command-parser.js

```diff
@@ -133,6 +135,248 @@ function canTalk(user, room, connection, message, targetUser) {
 	return true;
 }
 
+var CommandContext = exports.Context = (function () {
+	function CommandContext (options) {
+		this.target = options.target;
+		this.room = options.room;
+		this.user = options.user;
+		this.connection = options.connection;
+		this.cmd = options.cmd;
+		this.message = options.message;
+
+		this.levelsDeep = options.levelsDeep;
+		this._cmdMain = options._cmdMain;
+
+		this.cmdToken = options.cmdToken;
+		this.fullCmd = options.fullCmd;
+		this.originalMessage = options.originalMessage;
+		this.namespaces = options.namespaces;
+		this.commands = options.commands;
+	}
+
+	CommandContext.prototype.levelsDeep = 0;
```

Comment:
Wouldn't it be better to have default values in the constructor, rather than in the prototype? (This also avoids problems if we want to add an object later on...)